guides/0000-atomic-swap.md

```diff
@@ -0,0 +1,97 @@
+### What is atomic swap
+
+Atomic swap is a feature where two different users can exchange different tokens in the same transaction without any third party and any trust.
+
+For instance, imagine you want to buy an NFT and pay with HTR. This can be done in a single transaction with no trust between the buyer and the seller.
+
+### Operations flow
+
+In the example described here we will assume that there is a central backend to coordinate the atomic swap. Let's assume we have two users (user1 and user2), each one with their own wallets (W1 and W2) in their devices and they want to swap token1 and token2, i.e. user1 will send token1 from W1 to W2 and user2 will send token2 from W2 to W1. See the image below.
+
+![atomic swap wallets drawio](https://user-images.githubusercontent.com/3298774/145630746-ca4c0011-2e01-4574-8143-f3c64c214f4e.png)
+
+The first thing to do is to define how many tokens will be exchanged, i.e. user1 will send X token1 to user2 and user2 will send Y token2 to user1. The atomic swap will happen in some steps that will be done in both wallets, where the users will need to exchange some information after each step. Below there is an image with each step and the explanation of them.
+
+#### Step 1
```

Comment:
Before explaining each step in details, I'd add a summary like this:

1. Wallet 1 and Wallet 2 agrees to exchange 3 X and 1 Y.
2. Wallet 2 sends its inputs and change outputs to wallet 2.
3. Wallet 1 generate an unsigned tx with all inputs and outputs.
4. Wallet 1 signs its inputs (in my example, TxA.inputs[0] only).
5. Note that the tx is not ready to be pushed yet.
6. Wallet 1 sends bytes(tx)  to Wallet 2.
7. Wallet 2 receives bytes(tx) , decodes it, and check whether the inputs and ouputs are correct.
8. Wallet 2 signs its inputs.
9. Now tx is ready to be pushed.
10. Wallet 2 pushes the transaction.
11. Wallet 1 receives the transactions and confirms the swap.

Maybe it should in a sequence diagram.

Reply:
Maybe add an example as well.

For example, let TxA be a transaction. Let X and Y be two different tokens. Then,

TxA.inputs[0] = 10 X (wallet 1)
TxA.inputs[1] = 5 Y (wallet 2)
TxA.outputs[0] = 3 X (wallet 2)
TxA.outputs[1] = 7 X (wallet 1, change)
TxA.outputs[2] = 1 Y (wallet 1)
TxA.outputs[2] = 4Y (wallet 2, change)

In this case, TxA is an atomic swap of 3 X and 1 Y from wallet 1 and wallet 2.

Reply:
I've added the summary but I've changed a bit. The example I decided to use is with a central coordinator because it's the request we had from a use case.

About the example, I guess it's confusing this way, we already have the code example in the reference level, I guess it's enough. What do you think?